src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -525,5 +649,260 @@ .maxstack  3
 ");
         }
 
+        [Fact]
+        public void TupleUsageWithoutTupleLibrary()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int, string a) x = (b: 1, ""hello"", 2);
+    }
+}
+";
+            // PROTOTYPE we should expect specific diagnostics
+            var comp = CreateCompilationWithMscorlib(source);
+            Assert.True(comp.GetDiagnostics().HasAnyErrors());
+        }
+
+        [Fact]
+        public void TupleWithDuplicateNames()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int a, string a) x = (b: 1, b: ""hello"", b: 2);
     }
 }
+" + trivial2uple + trivial3uple;
+        }
+
+        [Fact]
+        public void DefaultValueForTuple()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int a, string b) x = (1, ""hello"");
+        x = default((int, string));
+        System.Console.WriteLine(x.a);
+        System.Console.WriteLine(x.b ?? ""null"");
+    }
+}
+" + trivial2uple;
+
+            var comp = CompileAndVerify(source, expectedOutput: @"0
+null");
+        }
+
+        [Fact]
+        public void TupleWithDuplicateMemberNames()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int a, string a) x = (b: 1, c: ""hello"", b: 2);
+    }
+}
+" + trivial2uple + trivial3uple;
+
+            // PROTOTYPE we also expect duplicate member name error on the tuple type declaration
+            var comp = CreateCompilationWithMscorlib(source);
+            comp.VerifyDiagnostics(
+                // (6,50): error CS8202: Tuple member names must be unique.
+                //         (int a, string a) x = (b: 1, c: "hello", b: 2);
+                Diagnostic(ErrorCode.ERR_TupleDuplicateMemberName, "b").WithLocation(6, 50));
+        }
+
+        [Fact]
+        public void TupleWithReservedMemberNames()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int Item1, string Item3) x = (Item2: 1, Item4: ""hello"", Item3: 2);
+    }
+}
+" + trivial2uple + trivial3uple;
+
+            // PROTOTYPE we also expect reserved member name error on the tuple type declaration
+            var comp = CreateCompilationWithMscorlib(source);
+            comp.VerifyDiagnostics(
+                // (6,40): error CS8201: Tuple member name is disallowed at this position.
+                //         (int Item1, string Item3) x = (Item2: 1, Item4: "hello", Item3: 2);
+                Diagnostic(ErrorCode.ERR_TupleReservedMemberName, "Item2").WithLocation(6, 40),
+                // (6,50): error CS8201: Tuple member name is disallowed at this position.
+                //         (int Item1, string Item3) x = (Item2: 1, Item4: "hello", Item3: 2);
+                Diagnostic(ErrorCode.ERR_TupleReservedMemberName, "Item4").WithLocation(6, 50));
+        }
```

Comment:
We should have a test for (Item1: 1, Item1: 2) and for (Item2: 1, Item2: 2). It looks like current behavior should be good, but we need to have test coverage for scenarios like that.